src/arch/xtensa/include/arch/task.h

```diff
@@ -107,19 +107,21 @@ static inline void task_set_data(struct task *task)
 	switch (task->priority) {
 	case TASK_PRI_MED + 1 ... TASK_PRI_LOW:
 		irq_task = *task_irq_low_get();
-		dst = &irq_task->list;
 		break;
 	case TASK_PRI_HIGH ... TASK_PRI_MED - 1:
 		irq_task = *task_irq_high_get();
-		dst = &irq_task->list;
 		break;
 	case TASK_PRI_MED:
 	default:
+#ifdef CONFIG_TASK_HAVE_PRIORITY_MEDIUM
```

Comment:
For the switch statement it's better to split like
```
switch () {
#if MED
case HIGH:
case MED:
case LOW:
#else 
case HIGH:
case LOW
#endif 
default:
 trace_error();
 return error;
}
```
This way any unsupported cases like MED are reported and logged (meaning topology with medium level task will fail here rather than failing with a more complex scheduling issue).